specs/agents/tracing-distributed-tracing.md

```diff
@@ -20,3 +20,45 @@ implementing it when this was still a draft, we named the field `elasticapmtrace
 wait with the implementation of the `tracestate` field. We chose to avoid hyphens in the field name in order to reduce risk of breaking field name limitations, such as we encountered with some JMS clients.
 In order to make sure we are fully aligned, all agents are implementing the
 specification described in [this commit](https://github.com/w3c/trace-context-binary/blob/571cafae56360d99c1f233e7df7d0009b44201fe/spec/20-binary-format.md).
+
+#### Tracestate
+
+For our own `es` `tracestate` entry we will introduce a `key:value` formatted list of attributes.
+This is used to propagate the sampling rate downstream, for example.
+See the [sampling](tracing-sampling.md) specification for more details.
+
+The general `tracestate` format is:
+
+    tracestate: es=key:value;key:value...,othervendor=<opaque>
+
+For example:
+
+    tracestate: es=s:0.1,othervendor=<opaque>
+
+
+##### Validation and length limits
+
+The [`tracestate`](https://www.w3.org/TR/trace-context/#tracestate-header)
+specification lists a number of validation rules.
+In addition to that,
+there are specific rules for the attributes under the `es` entry.
+
+Agents MUST implement these validation rules when mutating `tracestate`:
+
+- The `tracestate` field may contain a maximum of 32 entries.
+  An entry consists of a vendor key, and an opaque vendor value.
+- Vendor keys (such as `es`) have a maximum size of 256 chars.
```

Comment:
Vendor _values_, right?

Reply:
Both keys and values:

https://www.w3.org/TR/trace-context/#key
https://www.w3.org/TR/trace-context/#value

Values are mentioned a few lines down.

Reply:
And validation is only relevant/required when we mutate the header, right? Trying to figure out the right place to do the validation.

Reply:
Yes, that's right. Right now I think there's very little we need to validate.

- We only need to _add_ a `tracestate` key at the root, so there won't be any existing tracestate to validate.
- We don't need to mutate `tracestate` downstream, so all we need to do is parse and validate the `es` entry if it exists.

See the paragraph at the end of this section which was intended to cover this - let me know if I can add anything to make it clearer.